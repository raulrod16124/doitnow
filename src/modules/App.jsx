import React from "react";
import { Provider } from "react-redux";
import { Home } from "./Home/Home";
import store from '../store/store';

import './styles/main.scss';

export default function App() {
  return (
    <Provider store={store}>
      <div className="App">
        <Home/>
      </div>
    </Provider>
  );
}
