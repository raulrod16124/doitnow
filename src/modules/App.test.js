import { render, screen } from "@testing-library/react";
import React from "react";

import App from "./App";

describe("Testing App Component", () => {
  let wrapper;

  beforeEach(() => {
    wrapper = render(<App />);
  });

  test("should render App component correctly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should find a img Role", () => {
    expect(screen.getByRole("img")).toBeInTheDocument();
  });

  test("should find a Login title", () => {
    expect(screen.getByText("Login")).toBeInTheDocument();
  });

  test("should find a textbox Role", () => {
    expect(screen.getByRole("textbox")).toBeInTheDocument();
  });

  test("should find a placeholder Email", () => {
    expect(screen.getByPlaceholderText("Email")).toBeInTheDocument();
  });

  test("should find a placeholder Password", () => {
    expect(screen.getByPlaceholderText("Password")).toBeInTheDocument();
  });

  test("should find a Role of link", () => {
    expect(screen.getByRole("link")).toBeInTheDocument();
  });

  test("should find a Role of Button", () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
  });

  test("should find a Role of Button", () => {
    expect(screen.getByRole("button")).toBeInTheDocument();
  });
});
