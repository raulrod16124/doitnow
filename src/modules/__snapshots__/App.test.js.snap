// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Component should render App component correctly 1`] = `
<Provider
  store={
    Object {
      "@@observable": [Function],
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <div
    className="App"
  >
    <AuthProvider>
      <ExperiencePoints />
      <BrowserRouter>
        <div
          className="main"
        >
          <Routes
            setNavVisibility={[Function]}
          />
        </div>
      </BrowserRouter>
    </AuthProvider>
  </div>
  <ConfirmationPropmt />
</Provider>
`;
